sshnode.go

```diff
@@ -103,22 +120,45 @@ func newCmdStrWithSource(cmd string, env []string) string {
 	return command
 }
 
-func (n *SSHNode) getClientAndSession() (*ssh.Client, *ssh.Session, error) {
-	var client *ssh.Client
-	var s *ssh.Session
-	var err error
-
-	// Retry few times if ssh connection fails
+func (n *SSHNode) connect() error {
 	for i := 0; i < MaxSSHRetries; i++ {
-		client, err = n.dial()
+		client, err := n.dial()
 		if err != nil {
 			time.Sleep(SSHRetryDelay)
 			continue
 		}
+		n.client = client
+	}
+	if n.client == nil {
+		return fmt.Errorf("failed to connect to host")
```

Comment:
Can the `err` var above be put into the function scope so it can be part of this error message?  That will help with debugging